### PR TITLE
Enable import attributes parsing by default

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | https://github.com/babel/website/pull/2974
| Any Dependency Changes?  |
| License                  | MIT

It got consensus for Stage 4 at the October 2024 TC39 meeting.

This PR adds a `"deprecatedImportAssert"` parser plugin, as a replacement for `["importAttributes", { "depreatedAssertSyntax": true }]` so that we can remove the `importAttributes` plugin (because it won't be needed anymore).

This PR also has some changes relevant for Babel 8:
- Using the `importAssertions` parser plugin now throws, as it's removed and not supported anymore
- Using the `[importAttributes, { deprecatedAssertSyntax: true }]` parser plugin now throws, telling you to use `deprecatedImportAssert` instead (unless you are already using that plugin, in which case we simply ignore the old one)
- `@babel/generator` defaults to printing attributes as `with { x: "foo" }` instead of `with x: "foo"`.

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
This reached consensus for stage 4.

File: scripts/parser-tests/flow/allowlist.txt [hunk 1]
```diff
@@ -8,6 +8,7 @@ async_arrow_functions/with_type_parameters_types_disabled.js
 class_method_kinds/polymorphic_getter.js
 class_properties/migrated_0026.js
 comment_interning/declare_variable.js
+dynamic_import/migrated_0005.js
```
Comment (nicolo-ribaudo) [inline scripts/parser-tests/flow/allowlist.txt:11]: This is testing that dynamic import does not accept a second argument

**Comment by JLHwung** [comment]
Aside: Now that JSON module also reaches stage 4, should we rename `proposal-json-modules` and add it to preset-env?

**Comment by nicolo-ribaudo** [comment]
One day CI for this PR will be green, I promise.

**Comment by nicolo-ribaudo** [comment]
Ughh rebasing broke it

**Comment by nicolo-ribaudo** [comment]
The failure is a bug in Babel that, when fixed, triggers a bug in Node.js. I'll open a PR with a workaround soon.

**Review by liuxingbaoyu — APPROVED** [review]

File: packages/babel-parser/test/fixtures/es2020/dynamic-import/invalid-trailing-comma/output.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
```
[the remark this answers is not in the document]
Comment (nicolo-ribaudo) [reply packages/babel-parser/test/fixtures/es2020/dynamic-import/invalid-trailing-comma/output.json:1]: This can be just removed 👍

**Comment by nicolo-ribaudo** [comment]
https://github.com/babel/babel/pull/16928 fixes the Node.js 23 failure

**Review by JLHwung — APPROVED** [review]

**Pull request merged**